gnomad/utils/constraint.py

```diff
@@ -0,0 +1,113 @@
+# noqa: D100
+
+from typing import Union
+
+import hail as hl
+
+
+def annotate_mutation_type(
+    t: Union[hl.MatrixTable, hl.Table]
+) -> Union[hl.MatrixTable, hl.Table]:
+    """
+    Annotate mutation types.
+
+    The following annotations are added to the output Table:
+        - cpg
+        - transition
+        - mutation_type - one of "CpG", "non-CpG transition", or "transversion"
+        - mutation_type_model
+
+    :param t: Input Table or MatrixTable.
+    :return: Table with mutation type annotations added.
+    """
+    context_length = t.aggregate(hl.agg.mean(hl.len(t.context)))
```

Comment:
Is there a simpler way here to save more time?

Reply:
You can sample the first 100 values and base the length on that, then do something like this (and also import logging at the top of the script)
```
# Determine context length by sampling the first 100 values of 'context'
context_lengths = list(set((hl.len(ht.context).take(100))))
if len(context_lengths) > 1:
            raise ValueError("More than one length was found among the first 100 'context' values. Length of 'context' should be consistent.")
else:
    context_length = context_lengths[0]
    logger.info("Detected a length of %d for context length", context_length)

if context_length == 3:
        mid_index = 1
elif context_length == 7:
        mid_index = 3
else:
    raise ValueError(
        f"The length of context should be either 3 or 7, instead of {context_length}."
    )
```